File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.js
function solution(polynomial) {
    let x = 0;
    let num = 0;
    
    for (term of polynomial.split(" + ")) {
        if (term.includes("x")) {
            const constant = term.replace("x", "");
            x += constant === "" ? 1 : parseInt(constant);
        } else {
            num += parseInt(term);
        }
    }
    
    const result = [];
    if (x > 0) result.push(x === 1 ? "x" : `${x}x`);
    if (num > 0) result.push(num);
    return result.join(" + ").toString();
}

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    x = 0
    count = 0
    
    for term in polynomial.split(" + "):
        if "x" in term:
            replaced = term.replace("x", "")
            if replaced == "":
                x += 1
            else:
                x += int(replaced)
        else:
            count += int(term)
            
    answer = []
    
    if x == 1:
        answer.append("x")
    elif x > 1:
        answer.append(f"{x}x")

    if count > 0:
        answer.append(str(count))

    return " + ".join(answer)